### ISSUE TITLE
[DOTS-OCR]: Silent token divergence in TT SDPA decoder at seq≥2791 with bf16 KV cache

### ISSUE THREAD

**Issue by arobergeTT** [tenstorrent/tt-metal#42606]
### Component / Area

kernels

### Issue Type (optional)

Runtime Crash

### Observed

At `seq≥2791` with bf16 KV cache storage, the TT SDPA decoder silently diverges from the reference HF fp32 decoder. No assertion or warning is raised. The corruption compounds in autoregressive mode, producing garbage OCR output.

Concrete output observed on a Turkish legal document:

```
"tarihlihüküm, Davacı 'e ve davalalı 'e ... tevmişolduğu ... Teşmecin ... kesinleşstığı"
```

Per-step logit probe results (`probe_tt_decode_divergence.py`, HF teacher-forced vs. TT, same token sequence):
- **10 out of 179 decode steps** produce a different argmax token
- **First divergence at step 39:** HF predicts `'Mah'` (start of "Mahkememizden"), TT predicts `'\n\n'`
- **Critical divergence at steps 63–65:** HF produces `'hükü'→'ml'→'er'` ("hükümler"), TT produces `'h'→'m'→','` ("hm,")
- **Cosine similarity** between HF and TT logit vectors: 0.65–0.99 across steps (sufficient to flip argmax)

Config at time of failure:
- `seq_len`: 2791
- KV cache size: 2976 (tile-aligned)
- `WormholeComputeKernelConfig(HiFi4, fp32_dest_acc_en=True)`

Suspected root cause: bf16 KV-cache storage precision loss accumulated over 2791+ positions × 28 layers. In autoregressive (self-driving) mode, the first wrong token at step 39 compounds into further divergence.

### Expected

The model should either:
1. Produce token sequences matching the HF reference within an acceptable tolerance, or
2. Raise a warning that `seq≥2791` with bf16 KV cache exceeds the precision-safe operating range

### 1. Steps (exact commands)

Not provided. Model: `dots.mocr` (Qwen2.5-VL-derived), via `models/demos/qwen25_vl/tt/model.py`.

### 2. Input data / link or description

- Native-resolution scanned legal document producing `seq=2791`
- KV cache: 2976 (tile-aligned), bf16 storage
- `WormholeComputeKernelConfig(HiFi4, fp32_dest_acc_en=True)`
- Model: Qwen2.5-VL-derived (`dots.mocr`), 28-layer 3B decoder
- Repro scripts: `ocr_lab/probe_tt_decode_divergence.py`, `ocr_lab/run_hybrid_fast.py`

### 3. Frequency

Consistent — divergence at step 39 reproduced every run at `seq≥2791`. Also reproduced with `SDPAProgramConfig(exp_approx_mode=False)`, ruling out softmax approximation as the cause.

### 1. Software Versions

Not provided. Model: `dots.mocr` (Qwen2.5-VL-derived), via `models/demos/qwen25_vl/tt/model.py`.

### 2. Hardware Details

N150 - single-chip (1×1 mesh) used for this test.

### Is this a regression?

Unknown

### Logs & Diagnostics

No assertion or traceback produced. Divergence captured via `probe_tt_decode_divergence.py` (per-step argmax comparison + cosine similarity between HF and TT logit vectors).

Attempted workarounds that did **not** resolve the issue:
- `SDPAProgramConfig(exp_approx_mode=False)` — no effect
- fp32 KV cache — rejected by SDPA dtype assertion (`Data type must be BFLOAT16, BFLOAT8_B, or BFLOAT4_B`)
- `k_chunk_size<32` — rejected (`k_chunk_size must be divisible by TILE_SIZE`)

### Priority

None

### Impact

Silent numerical corruption on long-context decode (`seq≥2791`). Produces plausible-but-wrong OCR output with no error signal. Affects legal document OCR workloads where sequence lengths frequently exceed 2000 tokens.
